nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroComponent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URL;
+
+import org.apache.avro.Schema;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.runtime.RuntimeServiceException;
+import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * Avro component.
+ *
+ * @since 10.2
+ */
+public class AvroComponent extends DefaultComponent {
+
+    public static final String SCHEMA_XP = "schema";
+
+    public static final String FACTORY_XP = "factory";
+
+    public static final String REPLACEMENT_XP = "replacement";
+
+    private static final Log log = LogFactory.getLog(AvroComponent.class);
+
+    protected final AvroSchemaStoreService schemaStoreService;
+
+    protected final AvroFactoryService factoryService;
+
+    public AvroComponent() {
+        schemaStoreService = new AvroSchemaStoreServiceImpl();
+        factoryService = new AvroFactoryServiceImpl();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T getAdapter(Class<T> adapter) {
+        if (adapter.isAssignableFrom(schemaStoreService.getClass())) {
+            return (T) schemaStoreService;
+        } else if (adapter.isAssignableFrom(factoryService.getClass())) {
+            return (T) factoryService;
+        }
+        return null;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
+        if (SCHEMA_XP.equals(extensionPoint)) {
+            AvroSchemaDescriptor descriptor = (AvroSchemaDescriptor) contribution;
+            URL url = contributor.getRuntimeContext().getResource(descriptor.file);
+            try (InputStream stream = url == null ? null : url.openStream()) {
```

Comment:
may be use methods to make this readable,
usually the registration is lazy because EP can be overridden or disabled, only on component start descriptor are evaluated.

Reply:
I don't get it there.
The descriptor is used on registerContribution(), if I'm not wrong, it's the way descriptors are used everywhere.

Reply:
they are accumulated in registerContrib and processed (load class and so on) during service start

Reply:
Done